jina/types/document/__init__.py

```diff
@@ -457,8 +458,27 @@ def get_attrs(self, *args) -> Dict[str, Any]:
         .. seealso::
             :meth:`update` for bulk set/update attributes
 
+        .. note::
+            Arguments will be extracted using `dunder_get`
+            .. highlight:: python
+            .. code-block:: python
+
+                d = Document({'id': '123', 'hello': 'world', 'tags': {'id': 'external_id', 'good': 'bye'}})
+
+                assert d.id == '123'  # true
+                assert d.tags['hello'] == 'world' # true
+                assert d.tags['good'] == 'bye' # true
+                assert d.tags['id'] == 'external_id' # true
+
+                res = d.get_attrs(*['id', 'tags__hello', 'tags__good', 'tags__id'])
+
+                assert res['id'] == '123' # true
+                assert res['tags__hello'] == 'world' # true
+                assert res['tags__good'] == 'bye' # true
+                assert res['tags__id'] == 'external_id' # true
         """
-        return {k: getattr(self, k) for k in args if hasattr(self, k)}
+
+        return {k: getattr(self, k) if hasattr(self, k) else dunder_get(self._pb_body, k) for k in args}
```

Comment:
We should catch & log, when `dunder_get` fails here. Otherwise, one malformed document in the index can corrupt a lot of rankings/searches.

Reply:
the semantics are directly that it returns a None, I can make it fail if it makes us more secure

Reply:
The errors caused by a malformed document should not be caught by the `Document` type. Logging in a primitive type is also strange.

Reply:
`DocumentSet` does warning in `_extract_docs()`.

I am not sure what is the best option, so whatever we agree works for me.

What agreement we take @nan-wang @maximilianwerk ?

Reply:
I've no strong opinion of this. I'm ok with the following codes. @maximilianwerk 
```
            try:
                value = dunder_get(self._pb_body, k)
                ret[k] = value
                continue
            except (AttributeError, ValueError):
                defaultlogger.warning('some warning message')
                ret[k] = None
```

Reply:
Ah I am sorry, I should have read the tests in the first place. I believe it can only be an `AttributeError`, non? See
```
    with pytest.raises(AttributeError):
        d.get_attrs(*['inexistant'])
```

I'd rather catch it, to be honest.

Reply:
see the updated version